packages/function-tree/src/index.js

```diff
@@ -191,6 +191,11 @@ export class FunctionTree extends EventEmitter {
     this.runTree.on = this.on.bind(this)
     this.runTree.once = this.once.bind(this)
     this.runTree.off = this.removeListener.bind(this)
+
+    // When not extended
+    if (!returnInstance) {
```

Comment:
It is a good question indeed. The background for it is to allow an API that does this:

```js
import FunctionTree from 'function-tree'

const execute = new FunctionTree()

execute([])
```

It feels awkward to do:

```js
import FunctionTree from 'function-tree'

const tree = new FunctionTree()

tree.runTree([])
```

Cause you should not have access to the other stuff in there anyways. That said, I totally agree with you, it is not ideal... maybe we should rather change the name of the method:

```js
import FunctionTree from 'function-tree'

const tree = new FunctionTree()

tree.run([]) // Change to "run"
```

??